src/pages/permissions/panels/IdentityGroupsPanelConfirmModal.tsx

```diff
@@ -0,0 +1,108 @@
+import { ConfirmationModal, useNotify } from "@canonical/react-components";
+import { FC, useState } from "react";
+import { LxdIdentity } from "types/permissions";
+import {
+  generateGroupAllocationsForIdentities,
+  getChangesInGroupsForIdentities,
+  pivotIdentityGroupsChangeSummary,
+} from "util/permissionIdentities";
+import GroupsOrIdentityChangesTable from "./GroupOrIdentityChangesTable";
+import { updateIdentities } from "api/auth-identities";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { useToastNotification } from "context/toastNotificationProvider";
+import usePanelParams from "util/usePanelParams";
+
+interface Props {
+  onConfirm: () => void;
+  close: () => void;
+  selectedIdentities: LxdIdentity[];
+  addedGroups: Set<string>;
+  removedGroups: Set<string>;
+}
+
+const IdentityGroupsPanelConfirmModal: FC<Props> = ({
+  onConfirm,
+  close,
+  addedGroups,
+  removedGroups,
+  selectedIdentities,
+}) => {
+  const [submitting, setSubmitting] = useState(false);
+  const notify = useNotify();
+  const panelParams = usePanelParams();
+  const queryClient = useQueryClient();
+  const toastNotify = useToastNotification();
+
+  const identityGroupsChangeSummary = getChangesInGroupsForIdentities(
+    selectedIdentities,
+    addedGroups,
+    removedGroups,
+  );
+
+  const groupIdentitiesChangeSummary = pivotIdentityGroupsChangeSummary(
+    identityGroupsChangeSummary,
+  );
+
+  const handleSaveGroupsForIdentities = () => {
+    setSubmitting(true);
+
+    const newGroupsForIdentities = generateGroupAllocationsForIdentities(
+      addedGroups,
+      removedGroups,
+      selectedIdentities,
+    );
+
+    const payload = selectedIdentities.map((identity) => ({
+      ...identity,
+      groups: newGroupsForIdentities[identity.id],
+    }));
+
+    updateIdentities(payload)
+      .then(() => {
+        // modifying groups should invalidate both identities and groups api queries
+        void queryClient.invalidateQueries({
+          predicate: (query) => {
+            return [queryKeys.identities, queryKeys.authGroups].includes(
+              query.queryKey[0] as string,
+            );
+          },
+        });
+        const successMessage =
+          selectedIdentities.length > 1
+            ? `Updated groups for ${selectedIdentities.length} identities`
+            : `Updated groups for ${selectedIdentities[0].name}`;
+        toastNotify.success(successMessage);
+        panelParams.clear();
+        notify.clear();
+      })
+      .catch((e) => {
+        notify.failure("Update groups failed", e);
+      })
+      .finally(() => {
+        setSubmitting(false);
+        onConfirm();
+      });
+  };
+
+  return (
+    <ConfirmationModal
+      confirmButtonLabel="Confirm changes"
+      confirmButtonAppearance="positive"
+      onConfirm={handleSaveGroupsForIdentities}
+      close={close}
+      title="Confirm group modification"
```

Comment:
```suggestion
      title="Confirm modification"
```
As this can be about identities or groups, maybe keep this generic?

Reply:
same change in `GroupIdentitiesPanelConfirmModal`